packages/webview_flutter/webview_flutter_wkwebview/lib/src/web_kit_webview_widget.dart

```diff
@@ -140,18 +146,89 @@ class WebKitWebViewPlatformController extends WebViewPlatformController {
       if (!requiresUserAction) WKAudiovisualMediaType.none,
     };
   }
+
+  @override
+  Future<void> addJavascriptChannels(Set<String> javascriptChannelNames) async {
+    await Future.wait<void>(
+      javascriptChannelNames.where(
+        (String channelName) {
+          return !_scriptMessageHandlers.containsKey(channelName);
+        },
+      ).map<Future<void>>(
+        (String channelName) {
+          final WKScriptMessageHandler handler =
+              webViewProxy.createScriptMessageHandler()
+                ..setDidReceiveScriptMessage(
+                  (
+                    WKUserContentController userContentController,
+                    WKScriptMessage message,
+                  ) {
+                    javascriptChannelRegistry.onJavascriptChannelMessage(
+                      message.name,
+                      message.body!.toString(),
+                    );
+                  },
+                );
+          _scriptMessageHandlers[channelName] = handler;
+
+          final String wrapperSource =
+              'window.$channelName = webkit.messageHandlers.$channelName;';
+          final WKUserScript wrapperScript = WKUserScript(
+            wrapperSource,
+            WKUserScriptInjectionTime.atDocumentStart,
+            isMainFrameOnly: false,
+          );
+          webView.configuration.userContentController
+              .addUserScript(wrapperScript);
+          return webView.configuration.userContentController
+              .addScriptMessageHandler(
+            handler,
+            channelName,
+          );
+        },
+      ),
+    );
+  }
+
+  @override
+  Future<void> removeJavascriptChannels(
+    Set<String> javascriptChannelNames,
+  ) async {
+    if (javascriptChannelNames.isEmpty) {
+      return;
+    }
+
+    // WKWebView does not support removing a single user script, so this removes
+    // all user scripts and all message handlers and re-registers channels that
+    // shouldn't be removed. Note that this workaround could interfere with
+    // exposing support for custom scripts from applications.
+    webView.configuration.userContentController.removeAllUserScripts();
+    webView.configuration.userContentController
+        .removeAllScriptMessageHandlers();
+
+    javascriptChannelNames.forEach(_scriptMessageHandlers.remove);
+    final Set<String> remainingNames = _scriptMessageHandlers.keys.toSet();
+    _scriptMessageHandlers.clear();
+
+    return addJavascriptChannels(remainingNames);
```

Comment:
Nit: Since this is a (Future) void function, wouldn't it be more idiomatic to `await` rather than `return` here?